File: src/backend/music-server/music-api/src/main/java/com/lalala/music/dto/ArtistDTO.java
package com.lalala.music.dto;

import java.time.LocalDateTime;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import com.lalala.music.entity.ArtistEntity;
import com.lalala.music.entity.ArtistType;
import com.lalala.music.entity.GenderType;

@Getter
@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public final class ArtistDTO {
    private final Long id;
    private final String name;
    private final GenderType genderType;
    private final ArtistType type;
    private final String agency;
    private final LocalDateTime createdAt;
    private final LocalDateTime updatedAt;

    public static ArtistDTO from(ArtistEntity artist) {
        return new ArtistDTO(
                artist.getId(),
                artist.getName(),
                artist.getGender(),
                artist.getType(),
                artist.getAgency(),
                artist.getCreatedAt(),
                artist.getUpdatedAt());
    }
}

File: src/backend/streaming-server/src/main/kotlin/com/lalala/streaming/dto/ArtistDTO.kt
package com.lalala.streaming.dto

import java.time.LocalDateTime

data class ArtistDTO (
    val id: Long,
    val name: String,
    val genderType: GenderType,
    val type: ArtistType,
    val agency: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
) {}
